mmdet/models/roi_heads/roi_extractors/groie.py

```diff
@@ -0,0 +1,151 @@
+"""Generic RoI Extractor.
+
+A novel Region of Interest Extraction Layer for Instance Segmentation.
+"""
+
+from __future__ import division
+
+from mmcv.cnn import ConvModule
+from torch import nn
+
+from mmdet.core import force_fp32
+from mmdet.models.builder import ROI_EXTRACTORS
+from mmdet.ops import GeneralizedAttention, NonLocal2D
+from .single_level import SingleRoIExtractor
+
+
+class NoProcess(object):
+    """Apply identity function."""
+
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def __call__(self, x):
+        return x
+
+
+def SequentialModule(modules):
+    """Apply a sequential module."""
+    nn_modules = []
+    for mc in modules:
+        type_ = mc.pop('type')
+        nn_modules.append(load_processing_model(type_, mc))
+    return nn.Sequential(*nn_modules)
+
+
+# Models supported by GRoIE
+models = {
+    'NonLocal2D': NonLocal2D,
+    'GeneralizedAttention': GeneralizedAttention,
+    'ConvModule': ConvModule,
+    'Sequential': SequentialModule,
+    'ReLU': nn.ReLU,
```

Comment:
1. From the configs, it seems that 'ReLU' and `Sequential` are hardly used, do we need to keep them?
2. MMDetection V2.0 already supports [plugin](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/ops/plugin.py#L9), and it seems simply adding `ConvModule` into plugin could meet the needs in GROIE.

Reply:
1. Could you explain what you mean? thanks :smile: 
2. I'm rewriting this part to use plugins! :+1:

Reply:
For the first point, I mean that it seems `Sequential` or `NoProcess` seems to be not used in post_module or pre_modules. Therefore, it seems that we do not need to keep them as an option.

Reply:
removed. :smile: